### PR TITLE
Adjust new history navigation header

### PR DESCRIPTION
**Pull request by guerler**
This PR simplifies the navigation header by reducing the number of `dropdowns` from two to one. This reduces the number of clicks for 1) creating a new history and 2) switching between histories, which are likely the most common operations. It's also more in alignment with the currently existing history.

This PR replaces the panel dropdown icons in both, the history and the tool panel, with the default caret. Also panel buttons are now grouped in both, highlighted consistently on hover, and display similar tips. Additionally this PR contains minor adjustments to the tool panels grouping dropdown, the header size in the dropdown has been reduced and a check mark indicates the currently selected grouping option.

New History Navigation Header:

image

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Review by davelopez — COMMENTED** [review]
I really like the unified look of the two side panels now! :+1: 

I agree that Grids need to be Vueified and we must avoid duplication and displaying the *Saved Histories* from the navigation header seems right. However, to reduce the number of clicks for switching histories we can also put the switch icon directly in the navigation bar. I'm not against removing the *Switch History modal* if people think is really duplication, but I kind of liked it and I think it has its value because, in my non-power Galaxy user opinion, it's slightly more intuitive to switch histories and doesn't interfere so much with what you have in the center panel. For example, you may be trying to run a workflow and decide to change the current history to run it on different data, AFAIK now you'll have to kind of break your flow by replacing your central panel with the saved histories before going back.
I see both mechanisms as useful, one for quickly switching histories (simple and fast) and the other one a bit more exploratory (with more info per history).

I would like to know what others with more Galaxy day-by-day usage experience think about keeping or removing this modal.

**Comment by mvdbeek** [comment]
> I would like to know what others with more Galaxy day-by-day usage experience think about keeping or removing this modal.

I do think it is more useful than the backbone view for navigating histories, but I'm biased since I wrote it. To me the backbone view and the modal serve a different purpose: the backbone view is for managing histories, the modal is good at selecting a new history. That you're not destroying the current center panel is a nice benefit. I doubt any power users have seen the modal at this point.

I don't know if we should remove this before having vue-ified the backbone view (which could be put into a modal) ... it is clunky and very slow if you have large histories. Couldn't this be the basis for the backbone replacement ?

**Comment by guerler** [comment]
Thanks for the comments guys, so in my opinion, it is unfortunately not the basis for vueifying the grid. I know that this is not consensus but I think it would be much better to vuefiy grids generally and not attempt to rebuild every grid individually. The reason for that is simply the large number of grids, if we vuefiy them as one, all grids will become vuefied at the same time. Regarding showing the grid in a separate modal in order to avoid resetting the center panel, I think that we can still do that by just loading the grid into a modal. But again, I'd suggest to first vueify `grids-view.js`. Finally, since the beta history has not been activated yet, no one really should have seen the switching modal.

**Comment by bgruening** [comment]
> To me the backbone view and the modal serve a different purpose: the backbone view is for managing histories, the modal is good at selecting a new history. 

I agree with this in general. The current backbone view is more of a managing component. Reusing this to switch the history can interrupt the analysis flow, imho, as David pointed out. However, I do think the modal is only a slight improvement, as it requires an extra click etc. Imho what would be cool, and what would cover the majority of use cases is a "recent history"-widget. Selecting one of the last 5 (or 10?) histories covers probably a lot of use-cases. Can we have such a widget in the history + an entry `other histories`, which opens up the current modal / or backbone view (in that case I would not care much about interrupting the flow, assuming it will not be used much)?

**Comment by mvdbeek** [comment]
Why the limit to 5-10? the modal gets the data from vuex, so all the histories are there.That's also why it feels fast. The select that we had previously in the beta menu didn't offer enough enough metadata to be useful IMO (Update time and tags for how i use Galaxy). You can reduce the clicks by adding the modal directly to the history, and yeah switching to backbone with an extra button makes sense.

I don't care that much about the modal, but the replacement should be ready before removing it, and i think it'll take a bit more time, given  @guerler's comment:

I suspect we don't want to straight up port the grid, that doesn't really fit all the rest of the API, and i hope everyone agrees that the goal is to remove the old non-API controllers and move towards API endpoints. If there is disagreement on that point we should figure this out between the UI/UX and backend groups.

**Comment by bgruening** [comment]
> Why the limit to 5-10? the modal gets the data from vuex, so all the histories are there.

5-10 you can display in a normal drop-down as the first implementation of Mason. I think this was very convenient and fast for the user.

**Comment by guerler** [comment]
Yes, we all agree that we do not want to use the old non-API controllers and that we want to move towards API endpoints, that pretty obvious. The discussion is rather how to get there. I suggest to transform the js-code for grids first, this is pretty straightforward and will allow us to understand the ins and outs of the current functionality. We can then transition the resulting Vue components to assign api endpoints instead of legacy controller endpoints. This is most likely the best strategy to approach this. The alternative is to create new components directly associated to api endpoint which have some but not all functions of the already existing grids but duplicate code, while neither the previous nor the new code is sufficient.

**Comment by guerler** [comment]
I decided to restore the duplicate `saved histories` view for now in order to move forward with this PR, although I think that it will need to be replaced with the regular saved histories grid at one point. While I was doing that, I noticed that there seems to be a third variation of the `saved histories` table, which I removed. The problem with the alternative `saved histories` view is that it looks like a limited version of the original grid, its missing the number of files, the file status overview columns and the history size columns.

File: client/src/components/History/CurrentHistory/HistoryNavigation.vue [hunk 1]
```diff
@@ -6,17 +6,33 @@
...
+
+                <b-button
+                    v-b-modal.history-selector-modal
+                    data-description="show saved histories"
+                    size="sm"
+                    variant="link"
+                    v-b-tooltip.bottom.hover
+                    title="Show saved histories">
+                    <Icon fixed-width icon="list" />
+                </b-button>
```
Comment (davelopez) [inline client/src/components/History/CurrentHistory/HistoryNavigation.vue]: Maybe something like this to make it more obvious it is for quick changing the current History:
```suggestion
                <b-button
                    v-b-modal.history-selector-modal
                    data-description="switch to another history"
                    size="sm"
                    variant="link"
                    v-b-tooltip.bottom.hover
                    title="Switch to history">
                    <Icon fixed-width icon="exchange-alt" />
                </b-button>
```

**Review by davelopez — APPROVED** [review]
Thank you @guerler for considering keeping the **Quick switch to history** I still think it has value for the user and it's definitely fast for switching histories.

The **Saved Histories** grid has much more features and information (import new Histories, rename, delete, view the number of files, edit tags, multi-selection, share, copy, edit permissions...) that's why I think it serves a more "advanced" purpose than just switching. We can still keep the "Show saved histories" button next to the "Switch to history" one or if we don't want to clutter the navbar too much, move it inside the dropdown next to *Show Histories Side-by-Side* or just below *You have X histories* (which probably makes even more sense and otherwise the message doesn't have much to do with the current history context).

For avoiding duplication down the road, maybe we can add slots to reuse most of the grid features like searching or pagination but provide different rendering for the rows (simpler read-only for switching) or include additional options.

**Comment by dannon** [comment]
The 'history quick switcher' was a new feature specifically requested by users that serves a different purpose than the big grids, and I see it as a fundamentally different component, though of course they will share the histories store.

**Comment by guerler** [comment]
Thanks for all the reviews, there is a very clear vote in favor for the additional component and after my initial opposition, I read the comments and talked to @mvdbeek and finally came to a similar conclusion.

**Pull request merged**